pkg/gadgets/ebpftop/piditer/bpf/pid_iter.bpf.c

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright (c) 2020 Facebook, (c) 2022 The Inspektor Gadget authors */
+#include <vmlinux/vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "pid_iter.h"
+
+const volatile __u64 bpf_prog_fops_addr = 0;
+
+SEC("iter/task_file")
+int gadget_ebpftop(struct bpf_iter__task_file *ctx)
+{
+	struct file *file = ctx->file;
+	struct task_struct *task = ctx->task;
+	struct pid_iter_entry e;
+
+	if (!file || !task)
+		return 0;
+
+    // If we do have an address of bpf_prog_fops, use it; otherwise... well...
+    // TODO: Currently cilium/ebpf doesn't support .ksyms, this is why we get the info from userspace right now
+    if (bpf_prog_fops_addr != 0 && (__u64) file->f_op != bpf_prog_fops_addr)
+        return 0;
+
+	__builtin_memset(&e, 0, sizeof(e));
+	e.pid = task->tgid;
+	e.id = BPF_CORE_READ((struct bpf_prog *)(file->private_data), aux, id);
+
+	bpf_probe_read_kernel_str(&e.comm, sizeof(e.comm), task->group_leader->comm);
+	bpf_seq_write(ctx->meta->seq, &e, sizeof(e));
```

Comment:
I didn't know about `bpf_seq_write()`. Other gadgets use `bpf_seq_printf()`. Either is fine for me, but at some point we should probably be consistent. Using `bpf_seq_printf()` like other gadgets would have the advantage to not hardcode the size 24, or to convert the C string to a Go string.

Reply:
I did some performance tests and it seems that the `bpf_seq_printf()` variant is much slower. I don't think it matters too much (it's still pretty fast) in this case, but there might be other gadgets (in the future) that would profit from the `bpf_seq_write()` version. I replaced the hardcoded values and removed the "C" import, though.